subsys/net/ip/net_context.c

```diff
@@ -689,6 +689,49 @@ int net_context_put(struct net_context *context)
 	return 0;
 }
 
+static int bind_default(struct net_context *context,
+				    const struct sockaddr *dest_addr)
+{
+	struct net_if *iface = net_if_get_default();
```

Comment:
If we have multiple interfaces, then this selection might get wrong interface.
If the user has supplied a real destination address (and not just unspecified address), then we would need to select the interface according to that dest address. Only if the user does not set the dest address, then we can use the default interface. I wrote a small piece of code for that in my earlier review.